### ISSUE TITLE
Wrong report of "variable {0} is undeclared" using short obj literals

### ISSUE THREAD

**Issue by supersteves** [google/closure-compiler#1771]
I can't repro this in the debugger (can't guess what I need to configure to match my build), but here's what I'm seeing (names altered for brevity):

``` javascript
goog.module("a");
const f = v => {
  if (v instanceof Array) v.forEach(e => f(e));
  else console.log(v);
};
exports = {
  f
}
```

```
... ERROR - variable f$$module$a is undeclared ...
```

Debugging the compiler, the variable in scope is actually named:
``` module$contents$a_f ```

I'm experimenting with commonjs modules, if it's related.

The workaround is to use
``` javascript
exports = {
  f: f
}
```

**Comment by MatrixFrog** [comment]
Interesting. I assume this is with the compiler built from head, not the release? (If not, try it with a compiler built from head maybe...)

**Comment by supersteves** [comment]
Yes, 7a180297a38478d0e85c81d3266bcac1a31d32ea

**Comment by Dominator008** [comment]
@supersteves Can you provide a simple repro case, hopefully with the command line flags you used?

**Comment by supersteves** [comment]
Aha! I was experimenting with CommonJS modules and since gave up. 

Short syntax works fine without enabling processing of CommonJS modules.
But when I turn on ```CompilerOptions.processCommonJSModules```, it immediately fails.

No idea why - this is in a goog.module and it is being used by a goog.module...

I don't think I can currently create an online test case for CJS and ES6 modules. (Should I submit a PR with new enum entries in CompilationParam?) 

Can't put together an offline test, our build process is a Java wrapper with masses of complexity and the JS codebase is huge too. In short, we use advanced obfuscation and turn on every check and optimisation that doesn't look like an internal or experimental option.

**Comment by Dominator008** [comment]
@supersteves Hmm. I tried
```sh
java -jar build/compiler.jar --process_common_js_modules contents.js 
```
and it's still working. Haven't tried with a Java wrapper though.

**Comment by ChadKillingsworth** [comment]
There's tests for that: https://github.com/google/closure-compiler/blob/master/test/com/google/javascript/jscomp/ProcessCommonJSModulesTest.java#L276-L331

**Comment by supersteves** [comment]
@Dominator008 did you try *using* the export? It's the use of the module that fails, I'm getting ```module$contents$a_f``` just like you are, but the point of use is trying to call ```f$$module$a()```.

**Comment by ChadKillingsworth** [comment]
Ugh - I bet the commonjs module rewriting is interpreting goog.module files as common js. We need to fix that.

**Comment by jimmyt857** [comment]
I stumbled across this, and I think I am having the same issue. I am trying to include a common js module in my project, and after adding --process_common_js_modules to my command line I started getting errors.

I have constructed the following minimal example in which the error occurs.

foo.js:
```
goog.module('foo');
exports.x = 1;
```

main.js:
```
goog.module('testing');

const foo = goog.require('foo');
console.log(foo.x);
```

My command line:
```
npx google-closure-compiler --compilation_level='SIMPLE' --language_in='ES6_STRICT' --module_resolution=NODE --dependency_mode=STRICT --js='main.js' --js='foo.js' --js='node_modules/google-closure-library/**.js' --entry_point testing --js_output_file='out.js' --process_common_js_modules
node out.js
```

The output of all of this is ``undefined``. If I use @supersteves's workaround I get the expected result ``1``, although it comes with an annoying error:
```foo.js:5: WARNING - Suspicious re-assignment of "exports" variable. Did you actually intend to export something?```
